pegen/c_generator.py

```diff
@@ -145,6 +146,31 @@ class CParserGenerator(ParserGenerator, GrammarVisitor):
     def __init__(self, rules: Dict[str, grammar.Rule], file: Optional[IO[Text]]):
         super().__init__(rules, file)
         self.callmakervisitor = CCallMakerVisitor(self)
+        self._varname_counter = itertools.count()
+
+    def unique_varname(self, name="tmpvar"):
+        return name + "_" + str(next(self._varname_counter))
```

Comment:
I think we may want to use this more often when generating temporary varnames.

Reply:
Agreed, though I am not very familiar with `itertools.count()` so I'd rather use a simple counter that's updated here manually.